src/EditorFeatures/CSharpTest/DisposeAnalysis/DisposableFieldsShouldBeDisposedTests.cs

```diff
@@ -0,0 +1,1433 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.DisposeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using static Roslyn.Test.Utilities.TestHelpers;
+using Roslyn.Test.Utilities;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.DisposeAnalysis
+{
+    [Trait(Traits.Feature, Traits.Features.DisposeAnalysis)]
+    public sealed class DisposableFieldsShouldBeDisposedTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (new DisposableFieldsShouldBeDisposedDiagnosticAnalyzer(), null);
+
+        private Task TestDiagnosticsAsync(string initialMarkup, params DiagnosticDescription[] expectedDiagnostics)
+            => TestDiagnosticsAsync(initialMarkup, parseOptions: null, expectedDiagnostics);
+        private Task TestDiagnosticsAsync(string initialMarkup, CSharpParseOptions parseOptions, params DiagnosticDescription[] expectedDiagnostics)
+            => TestDiagnosticsAsync(initialMarkup, new TestParameters(parseOptions, retainNonFixableDiagnostics: true), expectedDiagnostics);
+        private Task TestDiagnosticMissingAsync(string initialMarkup, CSharpParseOptions parseOptions = null)
+            => TestDiagnosticMissingAsync(initialMarkup, new TestParameters(parseOptions, retainNonFixableDiagnostics: true));
+
+        [Fact]
+        public async Task DisposableAllocationInConstructor_AssignedDirectly_Disposed_NoDiagnostic()
+        {
+            await TestDiagnosticMissingAsync(@"
+using System;
+
+class A : IDisposable
+{
+    public void Dispose()
+    {
+    }
+}
+
+class B : IDisposable
+{
+    [|private readonly A a;|]
```

Comment:
what is this span represent?

Reply:
The test framework will look for any diagnostic that are reported by the analyzer that intersect this span. Analyzer only reports on the field symbol name, but I have kept larger annotation spans for tests validating no diagnostics reported. I verified that this test would fail if I remove dispose invocation on field 'a'